dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/collections/EvictingQueue.kt

```diff
@@ -0,0 +1,104 @@
+     * before the new element is added.
+     *
+     * @param element the element to be added.
+     * @return `true` if this collection changed as a result of the call
+     */
+    override fun add(element: T): Boolean {
+        if (size >= maxSize) {
+            removeFirst()
+        }
+        return super.add(element)
+    }
+
+    /**
+     * Adds all of the elements in the specified [elements] collection to the end of this queue.
+     *
+     * If the number of elements in [elements] is greater than or equal to [maxSize], the queue is cleared first,
+     * and only the last [maxSize] elements from [elements] are added.
+     *
+     * Otherwise, if adding [elements] would exceed the maximum capacity, the required number of oldest elements
+     * are evicted from the front of the queue to make room.
+     *
+     * @param elements the collection of elements to be added.
+     * @return `true` if the queue changed as a result of the call.
+     */
+    override fun addAll(elements: Collection<T>): Boolean {
+        if (elements.size >= maxSize) {
+            clear()
+            for ((index, element) in elements.withIndex()) {
+                if (index < elements.size - maxSize) continue
+                super.add(element)
+            }
+            return true
+        }
+
+        val spaceLeft = maxSize - size
+        for (index in 0 until elements.size - spaceLeft) {
+            removeFirst()
+        }
+
+        return super.addAll(elements)
+    }
+
+    /**
+     * This operation is not supported.
+     *
+     * Insertion at an arbitrary [index] is not allowed in [EvictingQueue] because it would break the
+     * FIFO (first-in, first-out) eviction policy.
+     *
+     * @param index the index at which the element is to be inserted.
+     * @param element the element to be inserted.
+     * @throws UnsupportedOperationException always.
+     */
+    override fun add(index: Int, element: T) {
+        throw UnsupportedOperationException("Insertion by index is not supported in EvictingQueue")
+    }
+
+    /**
+     * This operation is not supported.
+     *
+     * Insertion of a collection at an arbitrary [index] is not allowed in [EvictingQueue].
+     * The only allowed insertion is at the end of the queue (i.e., when [index] equals the current size).
+     *
+     * @param index the index at which to insert the first element from [elements].
+     * @param elements the collection of elements to be added.
+     * @return nothing; this method always throws an exception if [index] is not at the end.
+     * @throws UnsupportedOperationException if [index] is not equal to the current size of the queue.
+     */
+    override fun addAll(index: Int, elements: Collection<T>): Boolean {
+        if (index == size) return super.addAll(index, elements)
+        throw UnsupportedOperationException("Insertion by index is not supported in EvictingQueue")
+    }
```

Comment:
SDK shouldn't throw and crash the host app. Consider maybe logging instead?

Reply:
not relevant anymore